app/components/ui/Modal/Modal.jsx

```diff
@@ -0,0 +1,52 @@
+import React, { Fragment } from 'react';
+import PropTypes from 'prop-types';
+import { noop } from 'lodash';
+import Modal from 'react-modal';
+
+import styles from './Modal.scss';
+
+export const BaseModal = props => {
+  const {
+    backButtonText, closeModal, modalContent, isFullScreen,
+  } = props;
+
+  const modalBody = isFullScreen ? (
+    <Fragment>
+      <div className={styles.fullscreenModalHeader}>
+        <div className={styles.fullscreenModalBackButton} role="button" onClick={e => { e.stopPropagation(); closeModal(); }} tabIndex="0">
+          <i className="material-icons">keyboard_arrow_left</i>
+          {backButtonText}
+        </div>
+      </div>
+      <div className={styles.fullscreenModalBody}>
+        {modalContent}
+      </div>
+    </Fragment>
+  ) : null;
+
+  return (
+    <Modal
+      className={isFullScreen ? styles.fullScreenModalContent : styles.baseModalContent}
+      overlayClassName={styles.baseModalOverlay}
+      appElement={document.getElementById('root')}
+      {...props}
+    >
+      {modalBody}
+    </Modal>
+  );
+};
+
+BaseModal.propTypes = {
+  isOpen: PropTypes.bool.isRequired,
+  backButtonText: PropTypes.string,
+  closeModal: PropTypes.func,
+  modalContent: PropTypes.element,
+  isFullScreen: PropTypes.bool,
```

Comment:
Even if this isn't fully implemented yet, could you add some documentation on what the `isFullScreen` prop does? Your PR description of this was helpful for me, since I actually would have otherwise thought that `isFullScreen` would have referred to the "floating modal with overlay", since it almost cover the full screen. This is in contrast to modals I've seen on other sites which occupy a much smaller percentage of the screen.

Also, on mobile, do we always want to go full screen? If so, will setting the `isFullScreen` property have any effect?

Reply:
Ah ya, I'm looking at some of the designs that use that standard floating modal and it definitely doesn't take up as much of the screen as I have right now. Think I'm gonna just leave it as-is for now and we can style it better once we actually use it for the edit address modal. That's a good point about it transitioning to fullscreen on mobile. We don't have any mobile designs for this view ([figma here](https://www.figma.com/file/tBRClGsxYN9m6ZvL2STB6t/WIP-Edit-Redesign---Unscheduled?node-id=2003%3A2581)) but I'll check in with design when I implement it.

<img width="539" alt="Screen Shot 2020-05-31 at 2 32 41 PM" src="https://user-images.githubusercontent.com/7386336/83363118-cd45cd80-a34b-11ea-8053-e246fb06df65.png">

I'll add some documentation to the `isFullscreen` prop!